x/gov/types/proposal.go

```diff
@@ -8,38 +8,40 @@ import (
 
 	"gopkg.in/yaml.v2"
 
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
 // DefaultStartingProposalID is 1
 const DefaultStartingProposalID uint64 = 1
 
-// Proposal defines a struct used by the governance module to allow for voting
-// on network changes.
-type Proposal struct {
-	Content `json:"content" yaml:"content"` // Proposal content interface
-	ProposalBase
-}
-
 // NewProposal creates a new Proposal instance
-func NewProposal(content Content, id uint64, submitTime, depositEndTime time.Time) Proposal {
-	return Proposal{
-		Content: content,
-		ProposalBase: ProposalBase{
-			ProposalID:       id,
-			Status:           StatusDepositPeriod,
-			FinalTallyResult: EmptyTallyResult(),
-			TotalDeposit:     sdk.NewCoins(),
-			SubmitTime:       submitTime,
-			DepositEndTime:   depositEndTime,
-		},
+func NewProposal(content Content, id uint64, submitTime, depositEndTime time.Time) (Proposal, error) {
+	p := Proposal{
+		ProposalID:       id,
+		Status:           StatusDepositPeriod,
+		FinalTallyResult: EmptyTallyResult(),
+		TotalDeposit:     sdk.NewCoins(),
+		SubmitTime:       submitTime,
+		DepositEndTime:   depositEndTime,
+	}
+
+	msg, ok := content.(proto.Message)
+	if !ok {
+		return Proposal{}, fmt.Errorf("can't proto marshal %T", content)
```

Comment:
I don't think this is the correct error message to return in this case.

Reply:
What error message would you suggest?

Reply:
Actually this is fine. Ignore :)